runtime/src/message_processor.rs

```diff
@@ -1222,28 +1228,36 @@ impl MessageProcessor {
     ) -> Result<(), TransactionError> {
         let demote_sysvar_write_locks = feature_set.is_active(&demote_sysvar_write_locks::id());
         for (instruction_index, instruction) in message.instructions.iter().enumerate() {
+            let mut time = Measure::start("execute_instruction");
             let instruction_recorder = instruction_recorders
                 .as_ref()
                 .map(|recorders| recorders[instruction_index].clone());
-            self.execute_instruction(
-                message,
-                instruction,
-                &loaders[instruction_index],
-                accounts,
-                account_deps,
-                rent_collector,
-                log_collector.clone(),
-                executors.clone(),
-                instruction_recorder,
-                instruction_index,
-                feature_set.clone(),
-                bpf_compute_budget,
-                timings,
-                demote_sysvar_write_locks,
-                account_db.clone(),
-                ancestors,
-            )
-            .map_err(|err| TransactionError::InstructionError(instruction_index as u8, err))?;
+            let err = self
+                .execute_instruction(
+                    message,
+                    instruction,
+                    &loaders[instruction_index],
+                    accounts,
+                    account_deps,
+                    rent_collector,
+                    log_collector.clone(),
+                    executors.clone(),
+                    instruction_recorder,
+                    instruction_index,
+                    feature_set.clone(),
+                    bpf_compute_budget,
+                    timings,
+                    demote_sysvar_write_locks,
+                    account_db.clone(),
+                    ancestors,
+                )
+                .map_err(|err| TransactionError::InstructionError(instruction_index as u8, err));
+            time.stop();
+            if timings.collect_per_program_timings {
+                let program_id = instruction.program_id(&message.account_keys);
+                *timings.per_program_timings.entry(*program_id).or_insert(0) += time.as_us();
```

Comment:
In the case of same program being executed multiple times in a transaction, do we want to see the accumulated execution time in stats report, or perhaps averaged time? 
Cost model interested in later, if reporting accumulated time is desired, it is always possible to take average at the time of updating cost model.

Reply:
yes. I think we'll have to have a count stat here as well. I'll add that.